cmd/build.go

```diff
@@ -424,21 +431,99 @@ func buildNormal(targetOS string, vmArguments []string) {
 			}
 		}
 	}
+	if !buildOmitFlutterBundle {
+		var trackWidgetCreation string
+		if buildMode == build.DebugMode {
+			trackWidgetCreation = "--track-widget-creation"
+		}
 
-	outputEngineFile := filepath.Join(build.OutputDirectoryPath(targetOS), build.EngineFile(targetOS))
-	err = copy.Copy(
-		filepath.Join(engineCachePath, build.EngineFile(targetOS)),
-		outputEngineFile,
-	)
-	if err != nil {
-		log.Errorf("Failed to copy %s: %v", build.EngineFile(targetOS), err)
-		os.Exit(1)
-	}
+		cmdFlutterBuild := exec.Command(build.FlutterBin, "build", "bundle",
+			"--asset-dir", filepath.Join(build.OutputDirectoryPath(targetOS), "flutter_assets"),
+			"--target", buildTarget,
+			trackWidgetCreation,
+		)
+		cmdFlutterBuild.Stderr = os.Stderr
+		cmdFlutterBuild.Stdout = os.Stdout
 
-	err = copy.Copy(
-		filepath.Join(engineCachePath, "artifacts", "icudtl.dat"),
-		filepath.Join(build.OutputDirectoryPath(targetOS), "icudtl.dat"),
-	)
+		log.Infof("Bundling flutter app")
+		err = cmdFlutterBuild.Run()
+		if err != nil {
+			log.Errorf("Flutter build failed: %v", err)
+			os.Exit(1)
+		}
+		if buildMode.IsAot {
+			err := os.Remove(filepath.Join(build.OutputDirectoryPath(targetOS), "flutter_assets", "isolate_snapshot_data"))
+			if err != nil {
+				log.Errorf("Failed to remove unused isolate_snapshot_data: %v", err)
+				os.Exit(1)
+			}
+			err = os.Remove(filepath.Join(build.OutputDirectoryPath(targetOS), "flutter_assets", "vm_snapshot_data"))
+			if err != nil {
+				log.Errorf("Failed to remove unused vm_snapshot_data: %v", err)
+				os.Exit(1)
+			}
+			err = os.Remove(filepath.Join(build.OutputDirectoryPath(targetOS), "flutter_assets", "kernel_blob.bin"))
+			if err != nil {
+				log.Errorf("Failed to remove unused kernel_blob.bin: %v", err)
+				os.Exit(1)
+			}
+			dart := filepath.Join(engineCachePath, "dart"+build.ExecutableExtension(targetOS))
+			genSnapshot := filepath.Join(engineCachePath, "gen_snapshot"+build.ExecutableExtension(targetOS))
+			kernelSnapshot := filepath.Join(build.OutputDirectoryPath(targetOS), "kernel_snapshot.dill")
+			elfSnapshot := filepath.Join(build.OutputDirectoryPath(targetOS), "libapp.so")
+			cmdGenerateKernelSnapshot := exec.Command(
+				dart,
+				filepath.Join(engineCachePath, "gen", "frontend_server.dart.snapshot"),
+				"--sdk-root="+filepath.Join(engineCachePath, "flutter_patched_sdk"),
+				"--target=flutter",
+				"--aot",
+				"--tfa",
+				"-Ddart.vm.product=true",
+				"--packages=.packages",
+				"--output-dill="+kernelSnapshot,
+				buildTarget,
+			)
+			cmdGenerateKernelSnapshot.Stderr = os.Stderr
+			log.Infof("Generating kernel snapshot")
+			err = cmdGenerateKernelSnapshot.Run()
+			if err != nil {
+				log.Errorf("Generating kernel snapshot failed: %v", err)
+				os.Exit(1)
+			}
+			cmdGenerateAotSnapshot := exec.Command(
+				genSnapshot,
+				"--causal_async_stacks",
+				"--deterministic",
+				"--snapshot_kind=app-aot-elf",
+				"--strip",
```

Comment:
Do we want to `--strip` in Profile mode?

Reply:
Also some other flags need to be changes for profile mode. Can we ask the flutter team about that? In the mean time I would simply remove `--strip` for profile mode.